woocommerce/woocommerce
# [Experimental] Allow disabling product descriptions in the order summary.

### Changes proposed in this Pull Request:

Closes #51886

Introduce's a new block setting to the order summary items. It is disabled by default and also not available to enable unless you have the `experimental-blocks` flag enabled.

When the block setting toggle is enabled we do not show product descriptions in the order summary.



### How to test the changes in this Pull Request:

0. Enable experimental-blocks feature flag (should be enabled in dev by default), Have some products with a description
1. Now go to checkout page in editor
2. Within the block hierarchy select order summary items block (within order summary)
3. Note there should be a toggle available "Disable product descriptions"
4. Enable the toggle, see in the editor that product descriptions in the preview disappear
5. Save changes and view the front-end as a shopper and note that when you go to checkout your products do not show product descriptions.
6. Go back to editor and disable the toggle again and see that product descriptions are displayed (and in the frontend)

Before  | After
------------- | -------------
<img width="458" alt="Screenshot 2024-11-01 at 2 15 10 PM" src="https://github.com/user-attachments/assets/d2e2fb83-8a0a-42a9-ba8f-4aadf517d5ad">  | <img width="516" alt="Screenshot 2024-11-01 at 2 15 41 PM" src="https://github.com/user-attachments/assets/7f745e43-247d-44cb-8322-479da2aa9410">





### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
[Experimental] Allow disabling product descriptions in the product summary.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
